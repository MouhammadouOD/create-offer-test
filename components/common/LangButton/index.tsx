import React, { useContext } from 'react'
import { context } from '../../../contexts'
import ReactFlagsSelect from 'react-flags-select'

const LangButton = () => {
  const { locale, toggleLang } = useContext(context.LangContext)

  return (
    <>
      <ReactFlagsSelect
        className='menu-flags rfs_btn'
        selected={locale === 'en' ? 'GB' : 'FR'}
        onSelect={toggleLang}
        countries={['GB', 'FR']}
        selectedSize={13}
        showOptionLabel={false}
        showSelectedLabel={false}
      />
    </>
  )
}

export default LangButton
