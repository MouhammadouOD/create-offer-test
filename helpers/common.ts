const isArrayEmptyOrContainsEmptyObjects = (array: any) => {
  if (array?.length > 0) {
    return array?.every((item: any) => {
      return Object.keys(item).length === 0
    })
  }
  return true
}

export { isArrayEmptyOrContainsEmptyObjects }
