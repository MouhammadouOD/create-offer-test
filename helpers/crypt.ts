import { appEnv } from '../config'
import { AES, enc } from 'crypto-js'

const getCryptData = (data: any, isObject: boolean = false): string => {
  const dataHasCrypt = isObject ? JSON.stringify(data) : data
  const ciphertext = AES.encrypt(dataHasCrypt, appEnv.sessionSecret).toString()
  return ciphertext
}

const getDecryptData = (data: string, isObject: boolean = false): any => {
  var bytes = AES.decrypt(data, appEnv.sessionSecret)
  var decryptedData = isObject
    ? JSON.parse(bytes.toString(enc.Utf8))
    : bytes.toString(enc.Utf8)
  return decryptedData
}

export { getCryptData, getDecryptData }
