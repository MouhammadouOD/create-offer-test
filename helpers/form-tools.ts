const tranformDataToFormData = (
  datas: object,
  nameFile?: NameFile
): FormData => {
  let formData = new FormData()

  Object.entries(datas).forEach(([name, value]) => {
    let newValue
    switch (name) {
      case 'competences':
      case 'otherIndustries':
      case 'countries':
      case 'categories':
      case 'tags':
        newValue = JSON.stringify(value)
        break
      case 'deadlineEnd':
      case 'deadlineStart':
      case 'closingDate':
        newValue = value.toUTCString()
        break
      default:
        newValue = value ? value.toString() : null
        break
    }

    if (name === 'files' && nameFile && value) {
      for (const file of value) {
        formData.append(nameFile, file)
      }
    } else {
      newValue && formData.append(name, newValue)
    }
  })

  return formData
}

const tranformDataToSelectData = (
  datas: INameFrEn[],
  lang = 'fr'
): SelectOptionsForm[] => {
  const selectData: SelectOptionsForm[] = datas.reduce(
    (acc: SelectOptionsForm[], curr: INameFrEn) => {
      acc.push({
        value: curr.id,
        label:
          lang === 'fr' && curr?.nameFr
            ? curr.nameFr
            : curr.nameEn
            ? curr.nameEn
            : 'Error'
      })
      return acc
    },
    []
  )

  return selectData
}

const transformDataToReactSelectData = (
  datas: INameFrEn[],
  lang = 'fr'
): any => {
  const selectData: ReactSelectOption[] = datas.reduce(
    (acc: ReactSelectOption[], curr: INameFrEn) => {
      const name =
        lang === 'fr'
          ? curr.nameFr
            ? curr.nameFr
            : ''
          : curr.nameEn
          ? curr.nameEn
          : ''

      acc.push({
        value: name.toLowerCase(),
        label: name,
        id: curr.id
      })
      return acc
    },
    []
  )

  return selectData
}

export {
  tranformDataToFormData,
  tranformDataToSelectData,
  transformDataToReactSelectData
}
