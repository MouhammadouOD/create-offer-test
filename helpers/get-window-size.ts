import isIE from './is-ie'

function getSize(
  axis: number | string,
  body: any,
  html: any,
  computedStyle: any
) {
  return Math.max(
    body[`offset${axis}`],
    body[`scroll${axis}`],
    html[`client${axis}`],
    html[`offset${axis}`],
    html[`scroll${axis}`],
    isIE(10)
      ? parseInt(html[`offset${axis}`]) +
          parseInt(
            computedStyle[`margin${axis === 'Height' ? 'Top' : 'Left'}`]
          ) +
          parseInt(
            computedStyle[`margin${axis === 'Height' ? 'Bottom' : 'Right'}`]
          )
      : 0
  )
}

export default function getWindowSizes(document: {
  body: any
  documentElement: any
}) {
  const body = document.body
  const html = document.documentElement
  const computedStyle = isIE(10) && getComputedStyle(html)

  return {
    height: getSize('Height', body, html, computedStyle),
    width: getSize('Width', body, html, computedStyle)
  }
}
