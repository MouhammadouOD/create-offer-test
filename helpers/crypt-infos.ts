import * as cookie from 'cookie'
import { GetServerSidePropsContext } from 'next'
import { storageName } from '../constants'
import { getDecryptData, getCryptData } from './crypt'

type UserInfos = {
  token: string
  userId: number
  status: AccountStatusData
}

const getCryptUserInfos = (user: AccountData, token: string): string => {
  const userInfos = {
    token,
    userId: user.id,
    status: user.status
  }
  const value = getCryptData(userInfos, true)
  return value
}

const getDecryptUserInfos = (context: GetServerSidePropsContext) => {
  const parsedCookies = cookie.parse(
    context.req ? context.req.headers.cookie || '' : document.cookie
  )
  const userInfos: UserInfos = getDecryptData(
    parsedCookies[storageName.COOKIE_TOKEN],
    true
  )
  const { userId, token, status } = userInfos
  return { userId, token, status }
}

export { getCryptUserInfos, getDecryptUserInfos }
