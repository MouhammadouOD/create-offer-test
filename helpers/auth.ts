import { storageName, apiUrl, page } from '../constants'
import { webStorage } from '.'
import { api } from '../services'
import { addSession, deleteSession } from '../redux/session/sessionSlice'
import { makeStore } from '../store/app/store'
import Router from 'next/router'
import { deleteCookie, setCookie } from 'cookies-next'
import { getCryptUserInfos } from './crypt-infos'

const saveSession = (res: any) => {
  makeStore().dispatch(
    addSession({
      token: res.accessToken,
      tokenExpiresAt: res.accessTokenExpiresAt,
      me: res.user
    })
  )
  const userInfos = getCryptUserInfos(res.user, res.accessToken)
  const newToken = webStorage.saveInWebStorage(
    storageName.SESSION_ID,
    userInfos,
    false,
    'session',
    false
  )
  setCookie(storageName.COOKIE_TOKEN, newToken)
}

const logOut = async () => {
  makeStore().dispatch(deleteSession())
  deleteCookie(storageName.COOKIE_TOKEN)
  await api.fetchData(`${apiUrl.API_LOGOUT}`)
  webStorage.removeAllWebStorage()
  Router.push(page.PAGE_SIGNIN)
}

export { saveSession, logOut }
