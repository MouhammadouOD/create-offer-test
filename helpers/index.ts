import isBrowser from './is-browser'
import isIe from './is-ie'
import isValidEmail from './is-valid-email'
import setMaskEmail from './set-mask-email'
import getWindowSizes from './get-window-size'
import webStorage from './web-storage'
import dayjs from './daysjs-with-plugin'
import { tranformDataToFormData, tranformDataToSelectData } from './form-tools'
import { getCryptData, getDecryptData } from './crypt'
import { getCryptUserInfos, getDecryptUserInfos } from './crypt-infos'
import { isArrayEmptyOrContainsEmptyObjects } from './common'
import * as listPreviewFormData from './listPreviewFormData'
import reloadSession from './reload-session'
import * as offerHelpers from './offer'
import * as testUtils from './test-utils'
import * as messengerHelpers from './messenger-helpers'

export {
  isBrowser,
  isIe,
  isValidEmail,
  setMaskEmail,
  getWindowSizes,
  webStorage,
  dayjs,
  getCryptData,
  getDecryptData,
  tranformDataToFormData,
  tranformDataToSelectData,
  isArrayEmptyOrContainsEmptyObjects,
  listPreviewFormData,
  reloadSession,
  offerHelpers,
  testUtils,
  getDecryptUserInfos,
  messengerHelpers,
  getCryptUserInfos
}
