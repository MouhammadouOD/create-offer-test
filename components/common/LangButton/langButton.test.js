import React from 'react'
import { testUtils } from '../../../helpers'
import '@testing-library/jest-dom'
import LangButton from '.'

const { render, screen, renderer } = testUtils

describe('LangButton', () => {
  it('Should render Snapshot', async () => {
    const component = renderer.create(<LangButton />)
    expect(component).toMatchSnapshot()
  })
})
