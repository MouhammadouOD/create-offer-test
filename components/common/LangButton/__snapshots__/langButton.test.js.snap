// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangButton Should render Snapshot 1`] = `
<div
  className="ReactFlagsSelect-module_flagsSelect__2pfa2 menu-flags rfs_btn"
  data-testid="rfs"
>
  <button
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-labelledby="rfs-btn"
    className="ReactFlagsSelect-module_selectBtn__19wW7"
    data-testid="rfs-btn"
    disabled={false}
    id="rfs-btn"
    onClick={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "fontSize": 13,
      }
    }
    type="button"
  >
    <span
      className="ReactFlagsSelect-module_selectValue__152eS"
    >
      Select a country
    </span>
  </button>
</div>
`;
